### PR TITLE
Initialization ordering

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [X] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by kroening** [comment]
The actual change is small; look at 2nd commit.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d13f1da73134f583c9e97bd36bf40977d0accecf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96722472

File: src/linking/static_lifetime_init.cpp [hunk 1]
```diff
@@ -127,14 +127,25 @@ void static_lifetime_init(
...
   }
 
+  // first do framework variables
   for(const std::string &id : symbols)
-  {
-    auto code = static_lifetime_init(id, symbol_table);
-    if(code.has_value())
-      dest.add(std::move(*code));
-  }
+    if(has_prefix(id2string(id), CPROVER_PREFIX))
```
Comment (tautschnig) [inline src/linking/static_lifetime_init.cpp]: The use of `id2string` seems unnecessary, we are dealing with a `std::string` already.

**Review by tautschnig — APPROVED** [review]

File: src/linking/static_lifetime_init.cpp (hunk 1 above)
Comment (kroening) [reply src/linking/static_lifetime_init.cpp]: Fixed

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 958a61f6dcf17c7ac6640b9be0045348b3ff62c6).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96739026

**Pull request merged**